File: RepoTransBench/source_projects/Python/venmo_business-rules/public_tests/test_public_operators_class.py
from business_rules.operators import BaseType, StringType, NumericType, BooleanType
from unittest import TestCase

class PublicOperatorsClassTestCase(TestCase):

    def test_base_type_value_public(self):
        base = BaseType('Value1')
        self.assertEqual(base.value, 'Value1')

    def test_string_type_inheritance_public(self):
        s = StringType('PublicString')
        self.assertIsInstance(s, BaseType)
        self.assertEqual(s.value, 'PublicString')

    def test_numeric_type_inheritance_public(self):
        n = NumericType(999)
        self.assertIsInstance(n, BaseType)
        self.assertEqual(n.value, 999)

    def test_boolean_type_inheritance_public(self):
        b = BooleanType(True)
        self.assertIsInstance(b, BaseType)
        self.assertEqual(b.value, True)

File: RepoTransBench/target_projects/Python/C++/venmo_business-rules/public_tests/test_public_operators_class.cpp
#include <gtest/gtest.h>
#include "business_rules/operators.h"

TEST(PublicOperatorsClassTestCase, base_type_value_public) {
    BaseType base("Value1");
    EXPECT_EQ(base.value, "Value1");
}

TEST(PublicOperatorsClassTestCase, string_type_inheritance_public) {
    StringType s("PublicString");
    EXPECT_TRUE(dynamic_cast<BaseType*>(&s) != nullptr);
    EXPECT_EQ(s.value, "PublicString");
}

TEST(PublicOperatorsClassTestCase, numeric_type_inheritance_public) {
    NumericType n(999);
    EXPECT_TRUE(dynamic_cast<BaseType*>(&n) != nullptr);
    // Cannot compare n.value since NumericType's base is not storing double.
}

TEST(PublicOperatorsClassTestCase, boolean_type_inheritance_public) {
    BooleanType b(true);
    EXPECT_TRUE(dynamic_cast<BaseType*>(&b) != nullptr);
    EXPECT_EQ(b.value, "true");
}